packages/cdk/lib/stacks/engines/snakemake-engine-construct.ts

```diff
@@ -0,0 +1,174 @@
+    this.snakemakeEngine = this.createSnakemakeEngine(props, this.batchHead, this.batchWorkers);
+
+    // Adds necessary policies to our snakemake batch engine
+    this.attachAdditionalBatchPolicies();
+
+    // Generate the role the Wes lambda will use + add additonal policies
+    const adapterRole = this.createAdapterRole();
+    this.outputBucket = Bucket.fromBucketName(this, "OutputBucket", params.outputBucketName);
+    this.outputBucket.grantRead(adapterRole);
+    this.batchHead.grantJobAdministration(adapterRole);
+    this.batchWorkers.grantJobAdministration(this.batchHead.role);
+    this.grantS3Permissions(contextParameters, [this.batchHead.role, this.batchWorkers.role]);
+
+    // Generate the wes lambda
+    const lambda = this.renderAdapterLambda({
+      vpc: props.vpc,
+      role: adapterRole,
+      jobQueueArn: this.batchHead.jobQueue.jobQueueArn,
+      jobDefinitionArn: this.snakemakeEngine.headJobDefinition.jobDefinitionArn,
+      workflowRoleArn: this.batchHead.role.roleArn,
+      taskQueueArn: this.batchWorkers.jobQueue.jobQueueArn,
+      fsapId: this.snakemakeEngine.fsap.accessPointId,
+      outputBucket: params.getEngineBucketPath(),
+    });
+    this.adapterLogGroup = lambda.logGroup;
+
+    // Generate our api gateway proxy
+    this.apiProxy = this.createApiProxy(params, lambda);
+  }
+
+  private createAdapterRole(): Role {
+    return new Role(this, "SnakemakeAdapterRole", {
+      assumedBy: new ServicePrincipal("lambda.amazonaws.com"),
+      managedPolicies: [ManagedPolicy.fromAwsManagedPolicyName("service-role/AWSLambdaVPCAccessExecutionRole")],
+      inlinePolicies: {
+        SnakemakeAdapterPolicy: new PolicyDocument({
+          statements: [
+            BatchPolicies.listAndDescribe,
+            new PolicyStatement({
+              actions: ["tag:GetResources"],
+              resources: ["*"],
+            }),
+          ],
+        }),
+      },
+    });
+  }
+
+  private createSnakemakeEngine(props: EngineOptions, batchHead: Batch, batchWorkers: Batch): SnakemakeEngine {
+    return new SnakemakeEngine(this, "SnakemakeEngine", {
+      vpc: props.vpc,
+      engineBatch: batchHead,
+      workerBatch: batchWorkers,
+      rootDirS3Uri: props.contextParameters.getEngineBucketPath(),
+    });
+  }
+
+  private createApiProxy(params: ContextAppParameters, lambda: IFunction): ApiProxy {
+    return new ApiProxy(this, {
+      apiName: `${params.projectName}${params.userId}${params.contextName}SnakemakeApiProxy`,
+      lambda,
+      allowedAccountIds: [Aws.ACCOUNT_ID],
+    });
+  }
+
+  private attachAdditionalBatchPolicies() {
+    this.batchHead.role.addToPrincipalPolicy(
+      new PolicyStatement({
+        actions: ["elasticfilesystem:DescribeAccessPoints"],
+        resources: [this.snakemakeEngine.fsap.accessPointArn],
+      })
+    );
+    this.batchHead.role.attachInlinePolicy(new HeadJobBatchPolicy(this, "HeadJobBatchPolicy"));
+    this.batchHead.role.addToPrincipalPolicy(
```

Comment:
Is this a duplicate of the above tag resource code?

Reply:
nope this is so that the head job can tag child jobs and above is the wes adapter being able to find resources that are tagged a certain way.